python/calico/felix/profilerules.py

```diff
@@ -246,9 +258,11 @@ def _delete_chains(self):
         """
         # Need to block here: have to wait for chains to be deleted
         # before we can decref our ipsets.
-        self._iptables_updater.delete_chains(
-            self.iptables_generator.profile_chain_names(self.id),
-            async=False)
+        chain_names = self.iptables_generator.profile_chain_names(self.id)
+        if self._untracked:
```

Comment:
I did also like and consider encoding the rawness in TieredPolicyID - assuming that that is also what you mean here - but it seemed to me that there were places that construct a TieredPolicyID that do not have access to the policy content, and hence don't know if it's untracked or not.  Let me check...